app/bundles/EmailBundle/EventListener/ReportSubscriber.php

```diff
@@ -536,6 +539,51 @@ public function onReportGraphGenerate(ReportGraphEvent $event): void
                     );
                     break;
 
+                case 'mautic.email.graph.bar.read.clicked.unsubscribed.bounced':
+                    $this->addTrackableTablesForEmailStats($queryBuilder);
+                    $queryBuilder->select('e.id, e.name, e.sent_count, e.read_count, tr.unique_hits as `unique_clicks`,
+                        count(CASE WHEN '.self::DNC_PREFIX.'.id and '.self::DNC_PREFIX.'.reason = '.DoNotContact::UNSUBSCRIBED.' THEN 1 ELSE null END) as unsubscribed,
+                        count(CASE WHEN '.self::DNC_PREFIX.'.id and '.self::DNC_PREFIX.'.reason = '.DoNotContact::BOUNCED.' THEN 1 ELSE null END) as bounced'
+                    )
+                    ->groupBy('e.id');
```

Comment:
```suggestion
                    $queryBuilder->select('e.id, e.name, e.sent_count, e.read_count, tr.unique_hits as `unique_clicks`,
                        count(CASE WHEN '.self::DNC_PREFIX.'.id and '.self::DNC_PREFIX.'.reason = :unsubscribed THEN 1 ELSE null END) as unsubscribed,
                        count(CASE WHEN '.self::DNC_PREFIX.'.id and '.self::DNC_PREFIX.'.reason = :bounced THEN 1 ELSE null END) as bounced'
                    )
                    ->groupBy('e.id')
                    ->setParameter('unsubscribed', DoNotContact::UNSUBSCRIBED)
                    ->setParameter('bounced', DoNotContact::BOUNCED);
```
I would suggest to use the prepared statements

Reply:
Corrected.